src/components/Layouts/Primary/TopBar/TopBar.tsx

```diff
@@ -59,9 +59,11 @@ const TopBar = ({
               <img src={logo} alt="logo" style={{ cursor: 'pointer' }} />
             </NextLink>
           </Hidden>
-          <Box ml={10} flexGrow={1}>
-            <NavMenu />
-          </Box>
+          {accountListId && (
+            <Box ml={10} flexGrow={1}>
+              <NavMenu accountListId={accountListId} />
```

Comment:
I see that you're waiting to display the menu until an accountListId is loaded. I think you could still use the `useAccountListId` hook in the child component. It'd be one less prop. 🤷‍♂️ Guess it's fine either way.
```suggestion
              <NavMenu />
```